tests/test_host_check.py

```diff
@@ -2992,107 +2991,11 @@ def test_success(self, capsys):
             )
         assert textwrap.dedent(output) == textwrap.dedent(
             f"""\
-            PASS -- IOMMU
-                    IOMMU enabled for vfio-pci with the following PCI device(s):
-                    device1 (0000:00:00.0), device2 (0000:00:01.0), device3 (0000:00:02.0),
-                    device4 (0000:00:1f.2)
+            PASS -- IOMMU (IOMMU enabled for vfio-pci.)
             """
         )
         assert result is CheckState.SUCCESS
 
-    def test_no_iommu_devices(self, capsys):
-        """Test the case where no devices are found."""
```

Comment:
I don't see how these tests stub/mock what drivers are available either - i.e., there should be several cases here:

- vfio not present => NEUTRAL
- If vfio present then define "warning" as NEUTRAL if uio is present else WARNING in the following test cases:
  - `enable_unsafe_noiommu_mode` is Y => warning
  - For (`enable_unsafe_noiommu_mode` is N, `enable_unsafe_noiommu_mode` is not present):
    - `/sys/class/iommu/*/devices/*` glob is empty => warning
    - There's an error case that's probably not worth worrying about (not sure how `glob.glob` raises an Exception?
    - Else success.

Reply:
Actually I see now that what drivers are available is controlled by `cmd_effects`.  This is really quite hard to follow, but I think this is:

```
                cmd_effects=[
                    "",  # vfio-pci is supported
                    None,  # skip?????
                    subprocess.SubprocessError,  # uio is not supported?
                    subprocess.SubprocessError,  # not sure what this ever was
                    lshw_output,  # not relevant anymore
                ],

Reply:
for each check if a driver is supported, there can be up to two commands. The second is only called if the first fails so for vfio-pci in this case:
```
"", # => vfio-pci is supported
None, # this command is skipped
```
and for uio
```
subprocess.SubprocessError, # the first command returns an error
subprocess.SubprocessError, # so the second command is run, as both fail uio is not supported

Reply:
Ah gotcha